iOS/ExposureNotification/source/Extensions.cs

```diff
@@ -23,4 +26,42 @@ public static class ENRiskWeightRange {
 		public static byte Min { get; } = byte.MinValue;
 		public static byte Max { get; } = 100;
 	}
+
+	[Introduced (PlatformName.iOS, 12, 5)]
+	[Flags]
+	public enum ENActivityFlags : uint {
+		PeriodicRun = 1U << 2
+	}
+
+	public delegate void ENActivityHandler (ENActivityFlags activityFlags);
+
+	partial class ENManager {
+		[DllImport ("/usr/lib/libobjc.dylib", EntryPoint = "_Block_copy")]
+		private static extern IntPtr _Block_copy (ref BlockLiteral block);
+
+		[DllImport ("/usr/lib/libobjc.dylib", EntryPoint = "objc_msgSend")]
+		private extern static void void_objc_msgSend_IntPtr_IntPtr (IntPtr receiver, IntPtr selector, IntPtr arg1, IntPtr arg2);
+
+		private delegate void ENActivityHandlerCallbackDelegate (IntPtr block, ENActivityFlags activityFlags);
+
+		[MonoPInvokeCallback (typeof (ENActivityHandlerCallbackDelegate))]
+		private static unsafe void ENActivityHandlerCallback (IntPtr block, ENActivityFlags activityFlags)
+		{
+			var descriptor = (BlockLiteral*) block;
+			var del = (ENActivityHandler) (descriptor->Target);
+			del?.Invoke (activityFlags);
+		}
+
+		private static ENActivityHandlerCallbackDelegate ENActivityHandlerCallbackReference = ENActivityHandlerCallback;
+
+		public void SetLaunchActivityHandler (ENActivityHandler activityHandler)
```

Comment:
double check (test) with a `null` value - it might work (or throw)

If it throws then you might want to set `ptr = IntPtr.Zero;` instead of the `Block*` related code

Reply:
wow... 🤦

Reply:
Should I be running the CleanupBlock as well? Will that dispose anything that needs to stick around?

Reply:
Also, I see some code in xamarin-macios that does not do a _Block_copy. Should I be doing a copy?